src/table/components/head/HeadCellMenu.tsx

```diff
@@ -30,9 +31,13 @@ const MenuItem = ({
   sortDirection: string;
   sortFromMenu(evt: React.MouseEvent, sortOrder: string): void;
   setOpen: any;
+  isInteractionEnabled: boolean;
   isCurrentColumnActive: boolean;
 }) => {
-  const isDisabled = isCurrentColumnActive && sortOrder === (sortDirection === 'asc' ? 'A' : 'D');
+  let isDisabled = false;
```

Comment:
This logic seems unnecessary complex. A few remarks:
- Why not use `const isDisabled = ...`?
- Is `!isInteractionEnabled ? !isDisabled : isDisabled` not the same as `!isInteractionEnabled` ?
- Why is `sortOrder` defined as `'A' | 'B'` but sortDirection is `'asc' | 'desc'`?

Reply:
- For your first couple of questions you are right,  sorry I did a silly mistake here... I was thinking of `!isInteractionEnabled` as false but it's a conditional check that toggles the variable.
- column qSortIndicator in hypercube is defined in A, D,.. format. It is used in useSorting hook that also checks the menu item sortOrder somewhere in a condition, so both of them are set in a same format. Here sortDirection is a prop comes from the headCell parent and it is set in `asc`/`desc` format ... so I should find a way to convert it.

Reply:
I think we only need one of them (asc/desc or A/D). I would pass asc/desc then pass `sortDirection.charAt(0).toUpperCase()`

Reply:
I just looked into this. We need three versions of the sortDirections
- engine stores it using A/D and sort backend call expects it too
- mui `TableSortLabel` wants asc/desc as property to keep track of the icon (I have looked into just doing this manually and stop using that component, could be risky though)
- `aria-sort` needs it to be ascending/descending.
So I guess the typing should be better to reduce the confusion. I think the cleanest solution would be to store A/D on the column data. Then use enums to get asc and ascending, because I think they would only be used in TableHeadWrapper. Not sure how that impacts the virtual scrolling solution

Reply:
I tried your previous solution and it seems is working, have few conflicts need to fix them.  I also used Christofer's idea for setting a type to know which strings are accepted. Do you think you want them to be written with enum instead ?

Reply:
I think we can skip the enums, that would be an improvement for later